homeassistant/components/blebox/climate.py

```diff
@@ -0,0 +1,95 @@
+"""BleBox climate entity."""
+
+from homeassistant.components.climate import ClimateDevice
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+
+from . import BleBoxEntity, create_blebox_entities
+
+
+async def async_setup_entry(hass, config_entry, async_add):
+    """Set up a BleBox climate entity."""
+
+    create_blebox_entities(
+        hass, config_entry, async_add, BleBoxClimateEntity, "climates"
+    )
+
+
+class BleBoxClimateEntity(BleBoxEntity, ClimateDevice):
+    """Representation of a BleBox climate feature (saunaBox)."""
+
+    @property
+    def supported_features(self):
+        """Return the supported climate features."""
+        return SUPPORT_TARGET_TEMPERATURE
+
+    @property
+    def hvac_mode(self):
+        """Return the desired HVAC mode."""
+        return {None: None, False: HVAC_MODE_OFF, True: HVAC_MODE_HEAT}[
+            self._feature.is_on
+        ]
+
+    @property
+    def hvac_action(self):
+        """Return the actual current HVAC action."""
+        is_on = self._feature.is_on
+        if not is_on:
+            return None if is_on is None else CURRENT_HVAC_OFF
+
+        states = {None: None, False: CURRENT_HVAC_IDLE, True: CURRENT_HVAC_HEAT}
+
+        heating = self._feature.is_heating
+        return states[heating]
+
+    @property
+    def hvac_modes(self):
+        """Return a list of possible HVAC modes."""
+        return (HVAC_MODE_OFF, HVAC_MODE_HEAT)
```

Comment:
This should be a list

https://developers.home-assistant.io/docs/entity_climate/